src/providers/fireworks-ai/chatComplete.ts

```diff
@@ -0,0 +1,205 @@
+    default: false,
+  },
+  context_length_exceeded_behavior: {
+    param: 'context_length_exceeded_behavior',
+  },
+  user: {
+    param: 'user',
+  },
+};
+
+interface FireworksAIChatCompleteResponse extends ChatCompletionResponse {
+  id: string;
+  object: string;
+  created: number;
+  model: string;
+  usage: {
+    prompt_tokens: number;
+    completion_tokens: number;
+    total_tokens: number;
+  };
+}
+
+export interface FireworksAIValidationErrorResponse {
+  fault: {
+    faultstring: string;
+    detail: {
+      errorcode: string;
+    };
+  };
+}
+
+export interface FireworksAIErrorResponse extends ErrorResponse {}
+
+export interface FireworksAIStreamChunk {
+  id: string;
+  object: string;
+  created: number;
+  model: string;
+  choices: {
+    delta: {
+      role?: string | null;
+      content?: string;
+    };
+    index: number;
+    finish_reason: string | null;
+  }[];
+}
+
+export const FireworksAIErrorResponseTransform: (
+  response: FireworksAIValidationErrorResponse | FireworksAIErrorResponse
+) => ErrorResponse = (response) => {
+  if ('fault' in response) {
+    return generateErrorResponse(
+      {
+        message: response.fault.faultstring,
+        type: null,
+        param: null,
+        code: response.fault.detail.errorcode,
+      },
+      FIREWORKS_AI
+    );
+  }
+  return generateErrorResponse(response.error, FIREWORKS_AI);
+};
+
+export const FireworksAIChatCompleteResponseTransform: (
+  response:
+    | FireworksAIChatCompleteResponse
+    | FireworksAIValidationErrorResponse
+    | FireworksAIErrorResponse,
+  responseStatus: number
+) => ChatCompletionResponse | ErrorResponse = (response, responseStatus) => {
+  if (responseStatus !== 200) {
+    return FireworksAIErrorResponseTransform(
+      response as FireworksAIValidationErrorResponse | FireworksAIErrorResponse
+    );
+  }
+
+  if ('choices' in response) {
+    return {
+      id: response.id,
+      object: response.object,
+      created: response.created,
+      model: response.model,
+      provider: FIREWORKS_AI,
+      choices: response.choices.map((c) => ({
+        index: c.index,
+        message: {
+          role: c.message.role,
+          content: c.message.content,
+          tool_calls: c.message.tool_calls,
+        },
+        finish_reason: c.finish_reason,
+      })),
+      usage: {
+        prompt_tokens: response.usage?.prompt_tokens,
+        completion_tokens: response.usage?.completion_tokens,
+        total_tokens: response.usage?.total_tokens,
+      },
+    };
+  }
+  return generateInvalidProviderResponseError(response, FIREWORKS_AI);
+};
+
+export const FireworksAIChatCompleteStreamChunkTransform: (
+  response: string
+) => string = (responseChunk) => {
+  let chunk = responseChunk.trim();
+  chunk = chunk.replace(/^data: /, '');
+  chunk = chunk.trim();
+  if (chunk === '[DONE]') {
+    return `data: ${chunk}\n\n`;
+  }
+  const parsedChunk: FireworksAIStreamChunk = JSON.parse(chunk);
+  return (
```

Comment:
As per their docs, fireworks provides token usage in the last stream chunk for streaming mode. Can we also attach usage in our last transformed stream chunk. We already do something similar for bedrock stream transformer.

Reply:
done